src/libtego/source/ed25519.cpp

```diff
@@ -71,36 +71,60 @@ tego_v3_onion_service_id::tego_v3_onion_service_id(
     size_t serviceIdStringLength)
 {
     TEGO_THROW_IF_NULL(serviceIdString);
-    TEGO_THROW_IF_FALSE(serviceIdStringLength >= TEGO_V3_ONION_SERVICE_ID_LENGTH);
+    TEGO_THROW_IF_FALSE(is_valid(serviceIdString, serviceIdStringLength));
+
+    // copy to our internal buffer
+    std::copy(
+        &serviceIdString[0],
```

Comment:
correct, though overly verbose. You can also do:

```c++
std::copy(serviceIdString, serviceIdString + TEGO_V3_ONION_SERVICE_ID_LENGTH, this->data);
```

Reply:
though would prefer if you stuck the pointer and length into a std::string_view to avoid pointer match as much as possible